core/src/main/scala/dagr/core/cmdline/parsing/ClpReflectiveBuilder.scala

```diff
@@ -177,15 +171,15 @@ private[parsing] class ClpArgument(declaringClass: Class[_],
     */
   @SuppressWarnings(Array("unchecked"))
   def setArgument(values: String*) : Unit = {
-    (isFlag, isSetByUser, isCollection) match {
-      case (true, _, _) if values.isEmpty => this.value = true
-      case (_, true, _) => throw new IllegalStateException(s"Argument '$name' has already been set")
-      case (_, _, false) if values.size > 1 =>
-        throw new UserException(s"Argument '${this.names}' cannot be specified more than once.")
-      case _ =>
-        value = ParsingUtil.constructFromString(this.argumentType, this.unitType, values:_*)
-        this.isSetByUser = true
-    }
+    if (isSetByUser) throw new IllegalStateException(s"Argument '$name' has already been set")
+    if (!isCollection && values.size > 1) throw new UserException(s"Argument '${this.names}' cannot be specified more than once.")
+
+    if (isFlag && values.isEmpty)
+      this.value = true
+    else
+      this.value = ParsingUtil.constructFromString(this.argumentType, this.unitType, values: _*)
```

Comment:
I am usually open minded, but there are a few things about which I am not, and this is one of them.